src/App.tsx

```diff
@@ -1,26 +1,52 @@
-import React from 'react';
-import logo from './logo.svg';
+import { useState, useEffect } from 'react';
 import './App.css';
+import './normalize.css'
+import { Wine } from './interfaces/wine_interface';
+import { WineList } from './components/WinesList/WineList'
+import { fetchAllWines } from './api/wine_api'
 
-function App() {
+const App = () => {
+  const [wines, setWines] = useState<Wine[]>([])
+  const [isLoading, setIsLoading] = useState(true)
+  const [error, setError] = useState("")
+
+  useEffect(() => {
+    
+    (async () => {try {
```

Comment:
Something to consider here is how you want to handle errors. With your current approach, any time you need to fetch something you will need to wrap that in a try...catch and then handle the error in the catch. What other options do you have? What are the tradeoffs?

Reply:
I'll have to do some more research on this, but I can use error boundaries. I've never worked with it before, but it does sound interesting. Definitely worth looking into them and what the trade-offs are between them and using `try...catch`

Reply:
another cool option is to abstract this away with a custom hook. something like `useFetch` or whatever.